docs/rules/no-danger.md

```diff
@@ -24,6 +24,25 @@ var React = require('react');
 var Hello = <div>Hello World</div>;
 ```
 
+## Rule Options
+
+```js
+...
+"react/no-danger": [<enabled>, {
+  "checkCustomComponents": boolean,
+  "customComponentNames": Array<string>,
+}]
+...
+```
+
+### checkCustomComponents
+
+Defaults to `false`, when `enabled` allows the rule to check for custom components.
+
+### customComponentNames
+
+If you want to enable this rule for a specific set of custom components, you can pass `checkCustomComponents` as `true` and `customComponentNames` as the array of name of your custom components.
```

Comment:
i wonder if it'd be simpler to have just an array that defaults to `[]`, but can be set to `['*']` if you want to check everything?